File: src/test/java/io/rsocket/util/HeaderUtilTest.java
package io.rsocket.util;

import static org.junit.Assert.assertEquals;

import io.rsocket.cli.util.HeaderUtil;
import java.net.URISyntaxException;
import java.net.URL;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.Map;
import org.junit.Test;

public class HeaderUtilTest {
  @Test
  public void testSimpleValue() {
    assertEquals("hello", HeaderUtil.stringValue("hello"));
  }

  @Test
  public void testFileValue() throws URISyntaxException {
    assertEquals("value", HeaderUtil.stringValue("@" + path("value.txt")));
  }

  @Test
  public void headerMap() throws URISyntaxException {
    Map<String, String> map =
        HeaderUtil.headerMap(Arrays.asList("A: a", "B:b", "C: @" + path("value.txt")));

    assertEquals(3, map.size());
    assertEquals("a", map.get("A"));
    assertEquals("b", map.get("B"));
    assertEquals("value", map.get("C"));
  }

  @Test
  public void headerFileMap() throws URISyntaxException {
    Map<String, String> map = HeaderUtil.headerMap(Arrays.asList("@" + path("headers.txt")));

    assertEquals(2, map.size());
    assertEquals("a", map.get("A"));
    assertEquals("b", map.get("B"));
  }

  private String path(String resourceName) throws URISyntaxException {
    URL resource = getClass().getClassLoader().getResource(resourceName);
    return Paths.get(resource.toURI()).toFile().getPath();
  }
}

File: src/test/kotlin/io/rsocket/util/HeaderUtilTest.kt
package io.rsocket.util

import io.rsocket.cli.headerMap
import io.rsocket.cli.stringValue
import kotlinx.coroutines.runBlocking
import org.junit.Assert.assertEquals
import org.junit.Test
import java.net.URISyntaxException
import java.nio.file.Paths

class HeaderUtilTest {
  @Test
  fun testSimpleValue() {
    assertEquals("hello", stringValue("hello"))
  }

  @Test
  @Throws(URISyntaxException::class)
  fun testFileValue() {
    assertEquals("value", stringValue("@" + path("value.txt")))
  }

  @Test
  @Throws(URISyntaxException::class)
  fun headerMap() = runBlocking {
    val map = headerMap(listOf("A: a", "B:b", "C: @" + path("value.txt")))

    assertEquals(3, map.size.toLong())
    assertEquals("a", map["A"])
    assertEquals("b", map["B"])
    assertEquals("value", map["C"])
  }

  @Test
  @Throws(URISyntaxException::class)
  fun headerFileMap() = runBlocking {
    val map = headerMap(listOf("@" + path("headers.txt")))

    assertEquals(2, map.size.toLong())
    assertEquals("a", map["A"])
    assertEquals("b", map["B"])
  }

  @Throws(URISyntaxException::class)
  private fun path(resourceName: String): String {
    val resource = javaClass.classLoader.getResource(resourceName)
    return Paths.get(resource!!.toURI()).toFile().path
  }
}
